### PR TITLE
ESBJAVA-4942 : Adding time out equal to socket time out for write condition await in…

### PR DESCRIPTION
**Pull request by nwnpallewela**
… the flush content method

### CODE REVIEW

**Comment by vinok88** [comment]
Shall we mention the public Jira in the commit message.

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java [hunk 1]
```diff
@@ -82,13 +103,32 @@ public Pipe(IOControl producerIoControl, ControlledByteBuffer buffer,
...
+
+    /**
+     * Extract socket time out value from baseConfig and return default value if not found
+     *
+     * @param baseConfig
+     * @return
+     */
+    private long getSocketTimeOut(BaseConfiguration baseConfig) {
+        if (baseConfig instanceof SourceConfiguration) {
+            return ((SourceConfiguration) baseConfig).getHttpParams()
```
Comment (lankavitharana) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java]: Better make sure "((SourceConfiguration) baseConfig).getHttpParams()" cannot be null when it comes to this.

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java [hunk 2]
```diff
@@ -35,6 +40,8 @@
  */
 public class Pipe {
 
+    private static final int DEFAULT_TIME_OUT_VALUE = 180000;
```
Comment (isururanawaka) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java]: please add a comment here

File: modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java [hunk 3]
```diff
@@ -82,13 +104,36 @@ public Pipe(IOControl producerIoControl, ControlledByteBuffer buffer,
...
+        this.socketTimeOut = getSocketTimeOut(baseConfig);
+    }
+
+    /**
+     * Extract socket time out value from baseConfig and return default value if not found
+     *
+     * @param baseConfig
+     * @return
+     */
+    private long getSocketTimeOut(BaseConfiguration baseConfig) {
```
Comment (isudana) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java]: Can't we use an existing utility for this ?

Comment (nwnpallewela) [inline modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/Pipe.java]: Used existing one and made the socket time out value static.
Thanks

**Pull request merged**